gateway/middleware.go

```diff
@@ -301,6 +301,38 @@ func (t BaseMiddleware) UpdateRequestSession(r *http.Request) bool {
 	return true
 }
 
+// clearSession clears the quota, rate limit and complexity values so that partitioned policies can apply their values.
+// Otherwise, if the session has already a higher value, an applied policy will not win, and its values will be ignored.
+func (t BaseMiddleware) clearSession(session *user.SessionState) {
+	policies := session.PolicyIDs()
+	for _, polID := range policies {
+		t.Gw.policiesMu.RLock()
+		policy, ok := t.Gw.policiesByID[polID]
+		t.Gw.policiesMu.RUnlock()
+		if !ok {
+			continue
+		}
+
+		all := !(policy.Partitions.Quota || policy.Partitions.RateLimit || policy.Partitions.Acl || policy.Partitions.Complexity)
```

Comment:
can we rename `all` to `resetAll` to be more meaningful?

Reply:
Also, shall we try to add test case for this case as well?

Reply:
`all` is for all partitioned so it is nothing related to resetting.

Reply:
I have added the test.